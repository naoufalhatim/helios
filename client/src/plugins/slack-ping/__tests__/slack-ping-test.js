import React from 'react';
import renderer from 'react-test-renderer';
import SlackPing from '../slack-ping';
import moment from 'moment';

const data = {
  channel: '#general',
  pings: [
    {color: 'rgba(49, 163, 142, 1)', time: moment().startOf('day').add(1, 'minute').unix()},
    {color: 'rgba(237, 180, 49, 1)', time: moment().startOf('day').add(2, 'minute').unix()},
    {color: 'rgba(227, 21, 99, 1)', time: moment().startOf('day').add(3, 'minute').unix()}
  ]
};

jest.mock('react/lib/ReactDefaultInjection');

describe('SlackPing Component', () => {
  it('renders', () => {
    const slackPingOutput = renderer.create(
        <SlackPing {...data}/>
    );

    expect(slackPingOutput.toJSON()).toMatchSnapshot();
  });
});
