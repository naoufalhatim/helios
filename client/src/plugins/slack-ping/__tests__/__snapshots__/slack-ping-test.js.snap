exports[`SlackPing Component renders 1`] = `
<slack-ping>
  <span
    className="channel-name">
    ##general
  </span>
  <span>
    <div
      style={
        Object {
          "background": "rgba(49, 163, 142, 1)"
        }
      } />
    <div
      style={
        Object {
          "background": "rgba(237, 180, 49, 1)"
        }
      } />
    <div
      style={
        Object {
          "background": "rgba(227, 21, 99, 1)"
        }
      } />
  </span>
</slack-ping>
`;
